controllers/component_build_controller.go

```diff
@@ -201,9 +276,61 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	// Ensure component image repository credentials secret is linked to the pipeline service account
```

Comment:
Do we want to run this for each component change? How about limit to only when `image.redhat.com/image` changed?

Reply:
> Do we want to run this for each component change? How about limit to only when image.redhat.com/image changed? 

That's a good idea, but implementing that would require a new controller. Taking into account that switching to a new image repository needs a PR (in case of PaC) and PaC provision is done in this controller. So, I am not sure this approach gives us a gain, unfortunately.

> Or maybe skip when finalizer exists?

That would work if we hadn't supported switching of `ContainerImage` . Otherwise we need to unsure the link.
It does no harm if we don't read the secret itself.